docs/source/advanced.rst

```diff
@@ -0,0 +1,354 @@
+on a YAML OpenAPI specification which has been extended with any relevant
+OpenAlchemy extension properties.
+
+The :samp:`build_yaml` interface requires the :samp:`PyYAML` library to be
+installed. The :samp:`build_yaml` interface accepts the following arguments:
+
+* :samp:`spec_filename`: The name of the OpenAPI specification file. The file
+  must by a YAML file.
+* :samp:`package_name`: The name of the package to be produced. This will be
+
+  1. the name of the package if it is uploaded to a package index and
+  2. the name that is used to import the models after they have been installed).
+
+* :samp:`dist_path`: The directory to output the package files.
+* :samp:`format_`: Optionally, the format of the package to build:
+
+  * ``PackageFormat.NONE``: no package is built (default)
+  * ``PackageFormat.SDIST``: a sdist package is built
+  * ``PackageFormat.WHEEL``: a wheel package is built
+
+  The formats can be combined with the bitwise operator "or" (``|``), for
+  instance, building both sdist and wheel packages can be specified like this:
+
+  .. code-block:: python
+
+    format_=PackageFormat.SDIST|PackageFormat.WHEEL
+
+  .. warning::
+
+    In order to build a wheel distributable archive, the
+    `wheel <https://pypi.org/project/wheel/>`_ package is necessary.
+
+    It can be installed separately with ``pip install wheel``, or with
+    OpenAlchemy directly with ``pip install OpenAlchemy[wheel]``.
+
+.. _models-file:
+
+Models File
+-----------
+
+:samp:`OpenAlchemy` can optionally generate a file with all the
+:samp:`SQLAlchemy` models. Each model is constructed based on the
+:samp:`OpenApi` schema. The class inherits from the :samp:`SQLAlchemy` model
+defined on :samp:`open_alchemy.models`. The generated classes contain type
+information only, they do not provide any additional functionality on top of
+what the :samp:`SQLAlchemy` model provides. They are primarily
+used to enable IDE auto-complete and type hint support. The models can be used
+in the same way as the models that can be imported from
+:samp:`open_alchemy.models` and provide the full functionality of
+:samp:`SQLAlchemy` models.
+
+The following is a sample file generated for the above example:
+
+.. literalinclude:: ../../examples/simple/models_auto.py
+    :language: python
+    :linenos:
+
+The following information is recorded in the models file:
+
+* The name and type of each property of a schema.
+* The :ref:`from-dict` and :ref:`to-dict` function signatures, including
+  the type of the arguments and return values.
+* The :ref:`from-str` and :ref:`to-str` function signatures, including
+  the type of the arguments and return values.
+* The properties created on instance objects due to any :ref:`backref`.
+* Special SQLAlchemy properties for interacting with the database.
+* The object and property descriptions from the OpenAPI specification in the
+  class and function docstrings.
+
+.. _backrefs:
```

Comment:
This section on `backrefs` feels out of place here, should remove it to just before `.. _model-utilities:`?

Reply:
Not sure about that. to me it is not really a model utility either. It is more related to the extension properties I think. So maybe move it to an example instead?

Reply:
Reading it again, I think it is ok. It is a side note that explains a piece of internal functionality so might be misplaced in examples. Let's leave it here for now :smile: